pandas/core/arrays/datetimes.py

```diff
@@ -166,6 +176,9 @@ class DatetimeArrayMixin(dtl.DatetimeLikeArrayMixin):
     # by returning NotImplemented
     timetuple = None
 
+    # ensure that operations with numpy arrays defer to our implementation
+    __array_priority__ = 1000
```

Comment:
Are there tests in here that need this?

Reply:
Do we eventually want this on DatetimelikeArray (the most base class) so that all the arrays inherit the same value? It looks like PeriodArray doesn't define it or `__array_ufunc__` yet (there's an open issue IIRC).

I don't see any tests between `op(ndarray, datetimearray)`, but I may be missing them.